src/argilla/client/sdk/datasets/api.py

```diff
@@ -38,6 +38,28 @@ def get_dataset(client: AuthenticatedClient, name: str) -> Response[Dataset]:
     return _build_response(response=response, name=name)
 
 
+def list_datasets(client: AuthenticatedClient, workspace: Optional[str] = None) -> Response[List[Dataset]]:
+    url = "{}/api/datasets".format(client.base_url)
+
+    response = httpx.get(
+        url=url,
+        params={"workspace": workspace} if workspace else None,
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
```

Comment:
If the `rg.set_workspace` function is launched at some point. the `client.get_headers()` will contain the workspace header `"X-Argilla-Workspace"` which will be used to filter datasets. If you need to list all dataset available to user, you should remove the header in this call, and only keep the workspace logic through the query params.

```suggestion
    headers = client.get_headers().copy()
    headers.pop(_constans.WORKSPACE_HEADER_NAME, None)
    
    response = httpx.get(
        url=url,
        params={"workspace": workspace} if workspace else None,
        headers=headers,
        cookies=client.get_cookies(),
        timeout=client.get_timeout(),
    )
```